src/installer/corehost/corehost.cpp

```diff
@@ -200,23 +187,32 @@ int exe_start(const int argc, const pal::char_t* argv[])
 
     // Obtain the entrypoints.
     int rc;
-    hostfxr_main_startupinfo_fn main_fn_v2 = reinterpret_cast<hostfxr_main_startupinfo_fn>(pal::get_symbol(fxr, "hostfxr_main_startupinfo"));
-    if (main_fn_v2 != nullptr)
+#if defined(FEATURE_APPHOST)
+    if (bundle_marker_t::is_bundle())
     {
+        hostfxr_main_bundle_startup_info_fn main_fn_v3 = reinterpret_cast<hostfxr_main_bundle_startup_info_fn>(pal::get_symbol(fxr, "hostfxr_main_bundle_startup_info"));
+        if (main_fn_v3 == nullptr)
+        {
+            trace::error(_X("The required library %s does not support single-file apps."), fxr_path.c_str());
```

Comment:
What is the experience around this:
* I take a new (5.0) single-file (FDD) app and try to run it on a machine which doesn't have 5.0 installed.
* It fails with this error which doesn't really tell me anything. In addition, for GUI apps the reported `rc` is not recognized as special and there won't be any dialog.

Ideally we want experience similar to hostfxr missing or framework missing. I should get a URL to go to, and for GUI I should get a dialog.
Basically the problem is that this is effectively a version check, which happens before we run FX resolution. So we need it to behave in a very similar manner.

Reply:
I've fixed this, here's a sample output:
```
winhello.exe
The required library C:\Program Files\dotnet\host\fxr\5.0.0-preview.3.20169.1\hostfxr.dll does not support single-file apps.
  _ To run this application, you need to install a newer version of .NET Core.
  - https://aka.ms/dotnet-core-applaunch?missing_runtime=true&arch=x64&rid=win10-x64
```
![image](https://user-images.githubusercontent.com/8977622/78415750-c5caa800-75d8-11ea-83f5-87cefeaf3080.png)

Reply:
Can you please double-check that the URL which the dialog leads to contains apphost version information?

Reply:
Yes the link generated now is:
https://aka.ms/dotnet-core-applaunch?missing_runtime=true&arch=x64&rid=win10-x64&apphost_version=5.0.0-dev